File: zero/Maruko.Zero/Internal/Controllers/SysUserController.cs
﻿using Maruko.Core.Application;
using Maruko.Core.FreeSql.Internal.AppService;
using Maruko.Core.Web;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Cors;
using Microsoft.AspNetCore.Mvc;

namespace Maruko.Zero
{
    [EnableCors("cors")]
    [Route("api/v1/sysUsers/")]
    public class SysUserController : BaseCurdController<SysUser, SysUserDTO>
    {
        private readonly ISysUserService _user;

        [HttpPost("updatePersonalInfo")]
        public AjaxResponse<object> UpdatePersonalInfo(UpdatePersonalInfoRequest request)
        {
            return _user.UpdatePersonalInfo(request);
        }

        [HttpPost("auth/token")]
        [AllowAnonymous]
        public AjaxResponse<object> Login(LoginVM request)
        {
            return _user.Login(request);
        }

        [HttpPost("resetPassword")]
        public AjaxResponse<object> ResetPassword(ResetPasswordRequest request)
        {
            return _user.ResetPassword(request);
        }

        public SysUserController(ICurdAppService<SysUser, SysUserDTO> curd, ISysUserService user) : base(curd)
        {
            _user = user;
        }
    }
}

File: act/act-modules/act-modules-zero/src/main/java/com/act/modules/zero/internal/controllers/SysUserController.java
package com.act.modules.zero.internal.controllers;

import com.act.core.utils.AjaxResponse;
import com.act.core.utils.FriendlyException;
import com.act.core.web.BaseController;
import com.act.modules.zero.internal.application.user.SysUserService;
import com.act.modules.zero.internal.application.user.dto.LoginDTO;
import com.act.modules.zero.internal.application.user.dto.ResetPasswordRequest;
import com.act.modules.zero.internal.application.user.dto.SysUserDTO;
import com.act.modules.zero.internal.application.user.dto.UpdatePersonalInfoRequest;
import com.act.modules.zero.internal.domain.SysUser;
import com.act.modules.zero.internal.mapper.SysUserMapper;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@Api(tags = "系统用户")
@RestController
@RequestMapping("/api/v1/sysUsers/")
@SuppressWarnings("all")
public class SysUserController extends BaseController<SysUser, SysUserDTO, SysUserMapper> {

    @Autowired
    private SysUserService _user;

    @ApiOperation(value = "登录")
    @PostMapping("auth/token")
    public AjaxResponse<Object> login(@RequestBody LoginDTO request) throws InstantiationException, IllegalAccessException, FriendlyException {
        return _user.login(request);
    }

    @ApiOperation(value = "重置密码")
    @PostMapping("resetPassword")
    public AjaxResponse<Object> resetPassword(@RequestBody ResetPasswordRequest request) {
        return _user.resetPassword(request);
    }

    @ApiOperation(value = "修改个人信息")
    @PostMapping("updatePersonalInfo")
    public AjaxResponse<Object> UpdatePersonalInfo(@RequestBody UpdatePersonalInfoRequest request) {
        return _user.updatePersonalInfo(request);
    }
}
